sdk/keyvault/azure-keyvault-secrets/azure/keyvault/secrets/_shared/client_base.py

```diff
@@ -71,7 +72,7 @@ def __init__(self, vault_url, credential, **kwargs):
                 **kwargs
             )
             self._models = _KeyVaultClient.models(api_version=self.api_version)
-        except NotImplementedError:
+        except ValueError:
             raise NotImplementedError(
```

Comment:
Question here: the generated code has switched from raising a NotImplementedError to a ValueError if an unsupported API version is provided. I think either error is valid in this context, but I don't know if it's best to switch the error raised by the client for consistency, or to keep the NotImplementedError to avoid a (very small) breaking change. I marginally prefer the latter approach

Reply:
I agree. Either error seems reasonable to me, and consistency with prior versions of this library is more important to me than consistency with the generated code we wrap.